os-client/src/main/kotlin/io/provenance/scope/objectstore/client/OsClient.kt

```diff
@@ -76,10 +79,22 @@ open class OsClient(
             .let(customizeChannel)
             .build()
 
-        objectAsyncClient = ObjectServiceGrpc.newStub(channel)
-        objectFutureClient = ObjectServiceGrpc.newFutureStub(channel)
-        publicKeyBlockingClient = PublicKeyServiceGrpc.newBlockingStub(channel)
-        mailboxBlockingClient = MailboxServiceGrpc.newBlockingStub(channel)
+        //Mutate map of headers into appropriate Metadata construct
+        val headers = Metadata()
+            .also { metadata ->
+                extraHeaders.forEach { name, value ->
+                    metadata.put(
+                        Metadata.Key.of(name, Metadata.ASCII_STRING_MARSHALLER),
+                        value
+                    )
+                }
+            }
+
+        //Bind headers to stubs
+        objectAsyncClient = MetadataUtils.attachHeaders(ObjectServiceGrpc.newStub(channel), headers)
```

Comment:
I'm assuming it doesn't blow up or something if the headers object is empty?

Reply:
Yup. Works fine with or without